crates/ruff/src/rules/pylint/rules/yield_in_init.rs

```diff
@@ -9,6 +12,19 @@ use crate::{
 };
 
 define_violation!(
+    /// ### What it does
+    /// Checks for `__init__.py` methods that turned into generators
+    /// via the presence of `yield` or `yield from` statements.
+    ///
+    /// ### Why is this bad?
+    /// Generators are not allowed in `__init__.py` methods.
```

Comment:
Even without that mistake I find that description to be very bad.

Firstly it is misleading because generators can be used in `__init__` methods ... it's just that the `__init__` method cannot be a generator.

Secondly "is not allowed" is just completely nondescript ... what does that mean? Not allowed by whom?

I think [GitHub's CodeQL documentation](https://codeql.github.com/codeql-query-help/python/py-init-method-is-generator/) does a better job:

> The `__init__` method of a class is used to initialize new objects, not create them. As such, it should not return any value. By including a yield expression in the method turns it into a generator method. On calling it will return a generator resulting in a runtime error.

The last sentence here is the important part.

Reply:
Alright, I'll fix it up! Though, in my opinion, calling something "very bad" is not a productive way to deliver feedback and is a good way to turn a discussion into an argument.

Reply:
The documentation is written in a voice of authority ... when we use such a voice I think we should take great care that what we are writing is correct, clear and informative because otherwise we are very much bound to cause confusion or waste time of users who take our documentation at face value.

I think accurately and clearly answering `Why is this bad?` can be quite work-intensive for many of our lints. I'd still like us to put in that work rather than adding vague descriptions.

Reply:
I agree! Nowhere did I disagree that the documentation here could be improved.

Reply:
I thought a reference was sufficient but I don't want there to be any ambiguity so I'll just rewrite it entirely.